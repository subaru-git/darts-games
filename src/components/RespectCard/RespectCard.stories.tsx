/* eslint-disable import/named */
import React from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react'
import RespectCard from './RespectCard'

export default {
  title: 'Components/RespectCard',
  component: RespectCard,
} as ComponentMeta<typeof RespectCard>

const Template: ComponentStory<typeof RespectCard> = (args) => <RespectCard {...args} />

export const Default = Template.bind({})
Default.args = {}
