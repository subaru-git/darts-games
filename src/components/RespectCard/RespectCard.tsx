import React, { FC, Fragment } from 'react'
import { ExternalLinkIcon } from '@chakra-ui/icons'
import { Box, Center, Link, Image, Heading, Text } from '@chakra-ui/react'
import replace from 'react-string-replace'

type RespectCardProps = {
  data: WebsiteResult
}

const RespectCard: FC<RespectCardProps> = ({ data }) => {
  return (
    <Box borderRadius='xl' p={5} shadow='md' borderWidth='1px' maxWidth={'md'}>
      <Center>
        <Link href={data.url ?? '#'} isExternal>
          <Image
            src={data.image ?? ''}
            alt={`${data.title} image`}
            fallbackSrc='/fallback.webp'
            height='150px'
            maxWidth='400px'
          />
        </Link>
      </Center>
      <Heading as='h3' size={{ base: 'md', md: 'lg' }} py={2}>
        {data.title}
      </Heading>
      <Text fontSize={{ base: 'sm', md: 'md' }} wordBreak='break-word'>
        {replaceDescription(data.description ?? '')}
      </Text>
    </Box>
  )
}

const replaceDescription = (description: string) => {
  const urlReplaced = replace(
    description,
    /(https?:\/\/[\w/:%#\$&\?\(\)~\.=\+\-]+)/g,
    (match, i) => {
      return (
        <Fragment key={i}>
          <br />
          <Link href={match} isExternal color='teal.500'>
            {match} <ExternalLinkIcon />
          </Link>
          <br />
        </Fragment>
      )
    },
  )
  return urlReplaced
}

export default RespectCard
