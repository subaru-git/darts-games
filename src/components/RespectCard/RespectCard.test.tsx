import { render } from '@testing-library/react'
import RespectCard from './RespectCard'

test('should rendering', () => {
  const { container } = render(
    <RespectCard
      data={{
        title: 'title',
        image: 'image',
        description: 'description',
        url: 'url',
        type: 'type',
      }}
    />,
  )
  expect(container).toMatchSnapshot()
})
