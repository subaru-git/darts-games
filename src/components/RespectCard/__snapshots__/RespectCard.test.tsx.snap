// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div
    class="css-17of0ue"
  >
    <div
      class="css-gmuwbf"
    >
      <a
        class="chakra-link css-0"
        href="url"
        rel="noopener"
        target="_blank"
      >
        <img
          alt="title image"
          class="chakra-image__placeholder css-2y8fku"
          src="/fallback.webp"
        />
      </a>
    </div>
    <h3
      class="chakra-heading css-o5g2qp"
    >
      title
    </h3>
    <p
      class="chakra-text css-4h8fht"
    >
      description
    </p>
  </div>
</div>
`;
